bob-common/src/data.rs

```diff
@@ -4,11 +4,75 @@ use crate::{
 };
 use bob_grpc::{GetOptions, GetSource, PutOptions};
 use std::{
+    convert::TryInto,
     fmt::{Debug, Formatter, Result as FmtResult},
     hash::Hash,
 };
 
-pub type BobKey = u64;
+include!(concat!(env!("OUT_DIR"), "/key_constants.rs"));
+
+#[derive(Copy, Clone, Hash, PartialEq, Eq, Debug)]
+pub struct BobKey([u8; BOB_KEY_SIZE]);
+
+impl From<u64> for BobKey {
+    fn from(n: u64) -> Self {
+        let mut key = [0; BOB_KEY_SIZE];
+        key.iter_mut()
```

Comment:
you can make conversion just like this: `let key = n.to_be_bytes();`

Reply:
This does not work if BOB_KEY_SIZE != 8. Another solution:
```rust
impl From<u64> for BobKey {
    fn from(n: u64) -> Self {
        let mut key = n.to_be_bytes().to_vec();
        key.reverse();
        key.resize(BOB_KEY_SIZE, 0u8);
        key.reverse();
        Self(key.try_into().expect("key size mismatch"))
    }
}
```
Or use Little-Endian by default to avoid reverses.